### PR TITLE
SLCORE-927 read gitignore from bare repos too

### PR DESCRIPTION
**Pull request by serhat-yenican-sonarsource**
[SLCORE-927](https://sonarsource.atlassian.net/browse/SLCORE-927)

# For SonarSourcers:

- [x] Prefix the commit message with the ticket number, i.e. `SLCORE-XXXX` if you already have a ticket in Jira
- [ ] For standalone PRs without issue in Jira:
    - [ ] Mention Epic ID in this descrition to create a new Task in Jira
    - [ ] Mention Issue ID in this descrition to create a new Sub-Task in Jira
    - [ ] Do not mention any Jira issue to create a new Task in Jira without a parent
- [ ] When changing an API:
    - [ ] Explain in the JavaDoc the purpose of the new API
    - [ ] Document the change in [API_CHANGES.md](https://github.com/SonarSource/sonarlint-core/blob/master/API_CHANGES.md)
    - [ ] If the change breaks the current API, explicitly communicate those to the impacted consumers prior to merging (eg. IDE squad)
- [ ] Make sure the tests adhere to the convention:
    - [ ] All test method names should use `snake_case`, for example: `test_validate_input`.
- [x] Make sure checks are green: build passes, Quality Gate is green

# For external contributors:

In addition to the above, please review our [contribution guidelines](https://github.com/SonarSource/sonarlint-core/blob/master/docs/contributing.md) and ensure your pull request adheres to the following guidelines:

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed

[SLCORE-927]: https://sonarsource.atlassian.net/browse/SLCORE-927?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java [hunk 1]
```diff
@@ -62,6 +71,24 @@ class GitUtilsTest {
...
+  }
+
+  @AfterAll
+  @DisabledOnOs(OS.WINDOWS)
+  public static void afterAll() {
+    try {
+      FileUtils.forceDelete(bareRepoPath.toFile());
+      FileUtils.forceDelete(workingRepoPath.toFile());
+    } catch (Exception ignored) {
+      //It throws an exception in windows
```
Comment (kirill-knize-sonarsource) [inline backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java:89]: If it's disabled, you can just add an exception to the method signature.

File: backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java [hunk 2]
```diff
@@ -254,4 +281,51 @@ void should_consider_files_with_non_file_scheme_not_ignored() {
...
+    var e = assertThrows(IllegalStateException.class, () -> GitUtils.blameWithFilesGitCommand(bareRepoPath, emptySet));
+
+    assertThat(e).hasMessage("GitRepo is a bare repository");
+  }
+
+  private static void setUpBareRepo(Map<String, String> filePathContentMap) throws IOException, GitAPIException {
+    bareRepoPath = Files.createTempDirectory("bare-repo");
+    workingRepoPath = Files.createTempDirectory("working-repo");
+    // Initialize a bare repository
+    try (Git ignored = Git.init().setBare(true).setDirectory(bareRepoPath.toFile()).call()) {
```
Comment (kirill-knize-sonarsource) [inline backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java:307]: We usually use `var`. Same for several variables around this place.

**Review by kirill-knize-sonarsource — APPROVED** [review]
Minor neatpicks. Else LGTM.

File: backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java (hunk 1 above)
Comment (serhat-yenican-sonarsource) [reply backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java:89]: Ah, yes, that was my plan. I forgot to remove it. Thanks for the remainder.

Comment (serhat-yenican-sonarsource) [reply backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/GitUtilsTest.java:86]: Didn't work that way. I remove the annotation for disabling windows. Now the tests are green

**Pull request merged**